<!--Launch the entire SoFi autocontroller process/!-->

<launch>
    <arg name="auto_control" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="pitch_control" default="false" />
    <arg name="thrust_control" default="false" />

    <group if="$(arg auto_control)">
        <!--Launch the raspicam node/!-->
        <include file="$(find fishstatecontroller)/launch/camerav2_320x240.launch"/>
        <!--Launch the state estimation node/!-->
        <node type="BlobTracker_320x240.py" pkg="fishstatecontroller" name="state_estimation" output="screen" />
        <!--Launch the finite state machine node/!-->
        <node type="FishStateMachine.py" pkg="fishstatecontroller" name="finite_state_machine" output="screen">
	      <param name="heading_setpoint" type="double" value="0.0" />
	</node>
        <!--Launch the PID controller nodes/!-->
        <include file="$(find fishstatecontroller)/src/auto_controller.launch">
            <arg name="debug" value="$(arg debug)" />
            <arg name="pitch_control" value="$(arg pitch_control)" />
            <arg name="thrust_control" value="$(arg thrust_control)" />       
        </include>
    </group>

    <group unless="$(arg auto_control)">
        <!--Launch the manual controller node/!-->
        <node type="ManualController.py" pkg="fishstatecontroller" name="manual_controller" output="screen" />
    </group>

    <!--Launch the serial node/!-->
    <node type="serial_node.py" pkg="fishstatecontroller" name="serial" output="screen" />

</launch>
